contracts/bonding/BondingManager.sol

```diff
@@ -285,14 +265,30 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
     /**
      * @dev Withdraws withdrawable funds back to the caller after unbonding period.
      */
-    function withdraw() external afterInitialization whenSystemNotPaused currentRoundInitialized returns (bool) {
-        // Delegator must be unbonded
-        require(delegatorStatus(msg.sender) == DelegatorStatus.Unbonded);
+    function withdraw()
```

Comment:
So now we are overloading the withdraw() call with 2 meanings right?  If the delegator is still bonded, we just withdraw the unblondedAmount.  If it IS unbonded, we withdraw the bonded amount.

Is there a possibility where we have an unbonded delegator with both bonded AND unbonded amounts?

Reply:
Yeah...so the delegator would have to call withdraw twice: once to empty out the unbondedAmount and a second time to empty out the bondedAmount. A temporary better way could be: 

- If the delegator does not have an unbonded amount OR it is not unbonded
- If the delegator has an unbonded amount, empty it
- If the delegator is also unbonded, empty its bonded amount as well

Reply:
That makes sense.